cmd/thanos/query.go

```diff
@@ -76,6 +76,9 @@ func registerQuery(m map[string]setupFunc, app *kingpin.Application) {
 	stores := cmd.Flag("store", "Addresses of statically configured store API servers (repeatable). The scheme may be prefixed with 'dns+' or 'dnssrv+' to detect store API servers through respective DNS lookups.").
 		PlaceHolder("<store>").Strings()
 
+	rules := cmd.Flag("rule", "Addresses of statically configured rules API servers (repeatable). The scheme may be prefixed with 'dns+' or 'dnssrv+' to detect store API servers through respective DNS lookups.").
```

Comment:
Worth to mention that RulesAPI are now required to be storeAPI as well? (info endpoint needed).

Alternatively... We could define INFO endpoint for Rules as well (same as StoreAPI) and treat those separatedly. I think I kind of like this idea :thinking:

Reply:
I think the INFO endpoint idea is great. Generally as far as I understood we could leverage the INFO endpoint on RulesAPI to communicate back if this store endpoint actually is capable to serve RulesAPI. As far as I see we would have two cases:

- sidecar, rule: these components signal back the capability to server the RulesAPI.
- query, store, receive: these components signal back that they do not support the RulesAPI and always error out on the actual remote methods.

Does that sound reasonable at all?

Reply:
@bwplotka just to understand: if we don't want to use INFO endpoint for discoverability, what would be the benefit of implementing it redundantly?

Reply:
The only benefit would be: To have a service that ONLY implements RulesAPI and we can healthcheck it.